pkg/api/org_users.go

```diff
@@ -386,6 +389,15 @@ func (hs *HTTPServer) updateOrgUserHelper(c *contextmodel.ReqContext, cmd org.Up
 	if !c.OrgRole.Includes(cmd.Role) && !c.IsGrafanaAdmin {
 		return response.Error(http.StatusForbidden, "Cannot assign a role higher than user's role", nil)
 	}
+	// we do not allow to change role for external synced users
+	qAuth := login.GetAuthInfoQuery{UserId: cmd.UserID}
+	err := hs.authInfoService.GetAuthInfo(c.Req.Context(), &qAuth)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "Failed to get user auth info", err)
+	}
+	if qAuth.Result.AuthModule != "" && login.IsExternallySynced(hs.Cfg, qAuth.Result.AuthModule) {
```

Comment:
Missed it last review but shouldn't `IsExternallySynced` also confirm the provider is enabled in the cfg, so as to not block users from changing from provider -> no provider.

Behavior of when a provider is disabled should be to always return not ExternallySync

Reply:
Also occasion to fix `AuthProxtLabel` typo in `pkg/services/login/authinfo.go` 

EDIT: nit nit:`Github` -> `GitHub`

Reply:
the Github change will include all the settings as well. all of the settings are Github instead of GitHub. I'l see if I want to make that change here

Reply:
Yeah, do not bother with the GitHub if it's much work

Reply:
> Behavior of when a provider is disabled should be to always return not ExternallySync

this is true. I think we do not have all settings for that. Is it possible to add that in a separate PR?

Reply:
made it part of this PR - [85635cc](https://github.com/grafana/grafana/pull/64033/commits/85635cc1433a2fb8d5472958dff38eb26b106efb)